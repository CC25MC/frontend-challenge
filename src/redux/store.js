import { configureStore } from "@reduxjs/toolkit";
import { filesSlice } from "./slices";

export const store = configureStore({
  reducer: {
    files: filesSlice.reducer,
  },
});
