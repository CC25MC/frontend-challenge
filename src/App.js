import { useEffect } from "react";
import { NavbarB, TableB } from "./component";
import { useDispatch, useSelector } from "react-redux";
import { getFiles, getListFiles, getFile, setFile } from "./redux/slices";
import Spinner from "react-bootstrap/Spinner";
import Form from "react-bootstrap/Form";

const App = () => {
  const dispatch = useDispatch();
  const {
    isLoading,
    dataFiles = [],
    error = [],
    files = [],
    file,
  } = useSelector((state) => state.files);

  useEffect(() => {
    dispatch(getFiles());
    dispatch(getListFiles());
  }, [dispatch]);

  return (
    <>
      <NavbarB title="Full Stack Challenge" filesError={error} />
      <div className="mt-4 px-5">
        <Form.Select
          aria-label="selec file"
          value={file}
          onChange={(e) => {
            dispatch(setFile(e.target.value));
            dispatch(getFile(e.target.value));
          }}
        >
          <option value={"All"}>All</option>

          {files.map((item, key) => (
            <option key={key} value={item}>
              {item}
            </option>
          ))}
        </Form.Select>
        <div className="mt-3" />
      </div>

      {isLoading ? (
        <div
          style={{
            display: "flex",
            textAlign: "center",
            alignItems: "center",
            justifyContent: "center",
            width: "100%",
            height: "80vh",
          }}
        >
          <Spinner animation="grow" />
        </div>
      ) : (
        <div className="mt-4 px-5">
          <TableB data={dataFiles}/>
        </div>
      )}
    </>
  );
};

export default App;
