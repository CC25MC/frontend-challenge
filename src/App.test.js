import { render } from '@testing-library/react';
import App from './App';
import { Provider } from "react-redux";
import { store } from "./redux";

describe("App tests", () => {
  it("renders without crashing", () => {
    const component = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    component.getByText("Full Stack Challenge");
    expect(component.container).toMatchSnapshot();
  });
});

