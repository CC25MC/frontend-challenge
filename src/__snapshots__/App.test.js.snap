// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests renders without crashing 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      class="container"
    >
      <span
        class="navbar-brand"
      >
        Full Stack Challange
      </span>
    </div>
  </nav>
  <div
    class="mt-4 px-5"
  >
    <select
      aria-label="selec file"
      class="form-select"
    >
      <option
        value="All"
      >
        All
      </option>
    </select>
    <div
      class="mt-3"
    />
  </div>
  <div
    style="display: flex; text-align: center; align-items: center; justify-content: center; width: 100%; height: 80vh;"
  >
    <div
      class="spinner-grow"
    />
  </div>
</div>
`;
